### PR TITLE
e2e: use runId UUID for staticIPName to avoid collision

### PR DESCRIPTION
**Pull request by sjenning**
Fix for flake #20910

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -490,8 +493,14 @@ var _ = Describe("Services", func() {
 		if providerIs("gce", "gke") {
 			By("creating a static load balancer IP")
 			rand.Seed(time.Now().UTC().UnixNano())
-			staticIPName = fmt.Sprintf("e2e-external-lb-test-%d", rand.Intn(65535))
-			requestedIP, err = createGCEStaticIP(staticIPName)
+			// might need to try more than once for an unused staticIPName
+			for attempts := 0; attempts < maxAttempts; attempts++ {
+				staticIPName = fmt.Sprintf("e2e-external-lb-test-%d", rand.Intn(65535))
```
Comment (ixdy) [inline test/e2e/service.go]: maybe we should just instead be using `runId`, which is more unique? (not sure what the maximum length on this name can be, though.)

File: test/e2e/service.go [hunk 2]
```diff
@@ -54,6 +54,9 @@ const loadBalancerLagTimeout = 2 * time.Minute
 //TODO: once support ticket 21807001 is resolved, reduce this timeout back to something reasonable
 const loadBalancerCreateTimeout = 20 * time.Minute
 
+// How many attempts to retry on staticIPName collision
+const maxAttempts = 4
```
Comment (ixdy) [inline test/e2e/service.go]: should use a more specific name than `maxAttempts`, which is too generic. alternatively, maybe don't need to make this a separate constant.

File: test/e2e/service.go (hunk 1 above)
Comment (sjenning) [inline test/e2e/service.go]: Rules seem to be (from the GCE web console):
Name must start with a lowercase letter followed by up to 62 lowercase letters, numbers, or hyphens, and cannot end with a hyphen

Comment (sjenning) [inline test/e2e/service.go]: So "e2e-external-lb-test-" + 37 char UUID = 59.  So should be legal.

Comment (sjenning) [inline test/e2e/service.go]: @ixdy runId is private to util.go.  However, we could just call util.NewUUID() here to get a new one that has an equal chance of being unique.  That acceptable?

Comment (ixdy) [inline test/e2e/service.go]: Both service.go and util.go are in package e2e, so runId should be visible.

Comment (sjenning) [inline test/e2e/service.go]: ugh, still getting used to variable scopes in Go.  recovering C programmer here. new version is pushed.

**Comment by ixdy** [comment]
LGTM, thanks!

**Comment by ixdy** [comment]
ok to test

**Comment by thockin** [comment]
Thanks!

**Pull request merged**